hnn_core/parallel_backends.py

```diff
@@ -83,12 +83,13 @@ def __exit__(self, type, value, traceback):
 
     def _clone_and_simulate(self, net, trial_idx):
         # avoid relative lookups after being forked by joblib
-        from hnn_core.neuron import NeuronNetwork, _simulate_single_trial
+        from hnn_core.network_builder import NetworkBuilder
+        from hnn_core.network_builder import _simulate_single_trial
 
         if trial_idx != 0:
             net.params['prng_*'] = trial_idx
 
-        neuron_net = NeuronNetwork(net)
+        neuron_net = NetworkBuilder(net)
```

Comment:
I just wanted to confirm thoughts on naming convention here. In `examples/` there is no `neuron_net`, which is good because the user need not be aware that it has a collection of NEURON objects. Then inside `hnn_core` code the `neuron_net` is kept. This makes sense to me as a reminder that this object needs to be treated specially.

Reply:
ohhhh, I forgot to replace this :) Do you want me to keep it as is? No strong feelings ...

I think ultimately we want to have a single object with a method that builds in Neuron (see my email too you and @rythorpe yesterday for an example of this with the Cell class). That's the cleanest design in my opinion.

Reply:
I think keeping it as-is is fine for now.

I'm intrigued by your email. Will look at it and respond before Monday.

Reply:
> I just wanted to confirm thoughts on naming convention here. In `examples/` there is no `neuron_net`, which is good because the user need not be aware that it has a collection of NEURON objects. Then inside `hnn_core` code the `neuron_net` is kept. This makes sense to me as a reminder that this object needs to be treated specially.

This sounds fair to me as well. Thanks @jasmainak. I'll try to take a look at your email early tomorrow morning and get back to you.